File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> mp;
        for (auto num : nums) {
            mp[num]++;
        }
        int n = nums.size();
        int maj = 0;
        for (auto m : mp) {
            if (m.second > n / 2) {
                maj = m.first;
            }
        }
        return maj;
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer, Integer> mp = new HashMap<>();
        int n = nums.length;
        for(int num : nums){
            mp.put(num, mp.getOrDefault(num, 0) + 1);
            if(mp.get(num)>n/2)
                return num;
        }
        return -1;

    }
}